frame/clearing-house/src/lib.rs

```diff
@@ -93,14 +113,20 @@ pub mod pallet {
 
 	#[pallet::storage]
 	#[pallet::getter(fn get_initial_margin_ratio)]
+	#[allow(clippy::disallowed_types)]
 	/// Minimum margin ratio for opening a new position
 	type InitialMarginRatio<T: Config> = StorageValue<_, T::Decimal, ValueQuery>;
 
 	#[pallet::storage]
 	#[pallet::getter(fn get_maintenance_margin_ratio)]
+	#[allow(clippy::disallowed_types)]
 	/// Minimum margin ratio, below which liquidations can occur
 	type MaintenanceMarginRatio<T: Config> = StorageValue<_, T::Decimal, ValueQuery>;
 
+	#[pallet::storage]
+	/// Supported collateral asset ids
+	pub type CollateralTypes<T: Config> = StorageMap<_, Twox64Concat, AssetIdOf<T>, ()>;
```

Comment:
you could build permissioned core, and build on top of that permissioned list. not sure why doing permissions as part of this pallet. same of our Pablo(dexes). Core can always be permissionless, but router on top can give permissions. cc @hussein-aitlahcen

Reply:
You mean routing the transfer of assets from trader to clearing house through another pallet that checks if the collateral type can be deposited? That sounds interesting.